src/docfx/serve/Serve.cs

```diff
@@ -2,23 +2,66 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Diagnostics;
+using System.IO.Pipelines;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Nerdbank.Streams;
 
 namespace Microsoft.Docs.Build
 {
     internal static class Serve
     {
         public static bool Run(CommandLineOptions options, Package? package = null)
         {
+            if (options.Port == null)
+            {
+                throw new InvalidOperationException("`Port` parameter is required to serve curent docset as language server, please use `--port {port}`");
+            }
+
             if (!options.LanguageServer)
             {
                 Console.WriteLine("Docfx only support serving as a language server in 'Serve' mode, please use `--language-server`");
                 return true;
             }
 
-            GitUtility.ForceShellExecute = true;
+            var host = AspNetCore.WebHost.CreateDefaultBuilder()
```

Comment:
The `CreateDefaultBuilder` method does extra things we don't need like reading `appsettings.json`
```suggestion
            new WebHostBuilder()
                .UseKestrel()
```